### DISCUSSION TITLE
Matter support?

### DISCUSSION THREAD

**Discussion by stc1988** [Moddable-OpenSource/moddable#1170] [General]
Is there a plant to support Matter on [matter_esp32](https://github.com/Moddable-OpenSource/moddable/tree/matter_esp32) branch?
I' not familiar with Matter yet, but I'm interesting in device that works as home assistant.

**Comment by louisvangeldrop** [comment]
See https://github.com/project-chip/matter.js?tab=readme-ov-file#how-to-use-matterjs-in-own-projects for a Javascript solution

**Reply by phoddie** [reply]
Thanks for the link, @louisvangeldrop. That implementation is very cool. It is a great codebase for experimenting with Matter. We took a look at it a while back to see if it might work for embedded. Our conclusion was that there are a couple of significant issues. First is that it is really not written with resource constraints in mind. It could be a challenge to get it to fit. But, assuming we used a device with lots of RAM, the second challenge is dependencies. The one that stood out is Web Crypto, which matter.js uses quite heavily. Providing an implementation of that looks like a major undertaking, independent of the effort to bring-up matter.js.

**Comment by louisvangeldrop** [comment]
I believe that the matter.js project-group is working on a pure  Typescript implementation: 
https://github.com/orgs/project-chip/projects/11/views/1?pane=issue&itemId=40205892

**Comment by jonsmirl** [comment]
Any updates on Matter for Moddable? I tried running but I got a pile of errors on the imports; didn't work right with nested imports.

It would be very helpful to me to see an estimate for RAM and Flash consumption. Is a REPL environment feasible on an ESP32 S3 with 2MB PSRAM?  Similar to this: https://flic.io/flic-hub-sdk    That is Linux on an Allwinner R8.
